### PR TITLE
SI-6061 adds weakly conformance for number types to resolveOverloaded

### PR DESCRIPTION
**Pull request by dgruntz**
This fix changes `resolveOverloaded` in `scala.reflect.internal`
so that numeric widening on the actual argument types is also
considered for primitive number types. Needs changes in functions
applicable and mostSpecific.

Review: @cvogt @xeno-by

### CODE REVIEW

**Comment by adriaanm** [comment]
is this critical? (and should thus go into 2.10?)
the priority at https://issues.scala-lang.org/browse/SI-6061 says otherwise

(it's probably the latter that needs to be updated, but still)

**Comment by dgruntz** [comment]
I have changed the priority of the bug report (thought that the prioritization will be discussed/finalized next Tuesday).
Yes, I think it should go into 2.10. Is a new part in reflection (M5) and this fix enhances this new part.
Chris Hodapp will review the code (he is the author of resolveOverloaded), but he is on vacation now, but will definitely try to do something about this by next week.

-----Original Message-----

**Comment by adriaanm** [comment]
ok!

File: src/reflect/scala/reflect/internal/Symbols.scala [hunk 1]
```diff
@@ -93,6 +93,16 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
 
       // Begin Correlation Helpers
 
+      def weaklyConforms(a: Type, f: Type): Boolean = (a =:= f) || {
```
Comment (xeno-by) [inline src/reflect/scala/reflect/internal/Symbols.scala]: Can we use `weak_<:<` from `Types.scala` for that?

Comment (dgruntz) [inline src/reflect/scala/reflect/internal/Symbols.scala]: did not know about that. will change it. thanks!

Eugene Burmako reply@reply.github.com schrieb:

> @@ -93,6 +93,16 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
> 
> ```
>    // Begin Correlation Helpers
> ```
> -      def weaklyConforms(a: Type, f: Type): Boolean = (a =:= f) || {

Can we use `weak_<:<` from `Types.scala` for that?

---

Reply to this email directly or view it on GitHub:
https://github.com/scala/scala/pull/883/files#r1163832

**Comment by dgruntz** [comment]
simplified the implementation according to a hint proposed by @xeno-by: Made use of `weak_<:<` from `Types` instead of `<:<`.

**Comment by xeno-by** [comment]
lgtm

**Comment by clhodapp** [comment]
First: Github's comment submission system seems slow and buggy. I clicked the submit button three times because it was taking so long. It posted three times. I told it to delete two of them. It deleted all three.

Second: LGTM

Third: I believe that there are probably a number of improvements of this type (using existing support functions) that can be made to make this method better-behaved.

**Pull request merged**